code/online_challenges/src/project_euler/problem_003/problem_003.js

```diff
@@ -0,0 +1,13 @@
+let num = 600851475143;
+let highestPrimeFactor = 0;
+let currentFactor = 2;
+
+while (num !== 1) {
+  if (num % currentFactor === 0 && currentFactor > highestPrimeFactor) {
+    highestPrimeFactor = currentFactor;
```

Comment:
currentFactor could be not a prime, for example for num = 8, in the last iteration we get currentFactor = 4.

Reply:
You are right, I should reset the factor at some point.